src/index.js

```diff
@@ -1,4 +1,168 @@
 'use strict'
 
-exports.IPLDService = require('./ipld-service')
-exports.resolve = require('./resolve')
+const Block = require('ipfs-block')
+const pull = require('pull-stream')
+const traverse = require('pull-traverse')
+const utils = require('./utils')
+const CID = require('cids')
+const until = require('async/until')
+const IPFSRepo = require('ipfs-repo')
+const MemoryStore = require('../node_modules/interface-pull-blob-store/lib/reference.js')
+const BlockService = require('ipfs-block-service')
+
+const dagPB = require('ipld-dag-pb')
+const dagCBOR = require('ipld-dag-cbor')
+
+class IPLDResolver {
+  constructor (blockService) {
+    // nicola will love this!
+    if (!blockService) {
+      const repo = new IPFSRepo('in-memory', { stores: MemoryStore })
+      blockService = new BlockService(repo)
+    }
+
+    this.bs = blockService
+    this.resolvers = {}
+
+    // Support by default dag-pb and dag-cbor
+    this.support(dagPB.resolver.multicodec, dagPB.resolver, dagPB.util)
+    this.support(dagCBOR.resolver.multicodec, dagCBOR.resolver, dagCBOR.util)
+  }
+
+  // Adds support for an IPLD format
+  // default ones are dag-pb and dag-cbor
+  support (multicodec, resolver, util) {
+    this.resolvers[multicodec] = {
+      resolver: resolver,
+      util: util
+    }
+  }
+
+  resolve (cid, path, callback) {
+    if (path === '/') {
+      return this.get(cid, callback)
+    }
+
+    let value
+
+    until(
+      () => {
+        if (!path || path === '' || path === '/') {
+          return true
+        } else {
+          // continue traversing
+          if (value) {
+            cid = new CID(value['/'])
+          }
+          return false
+        }
+      },
+      (cb) => {
+        // get block
+        // use local resolver
+        // update path value
+        this.bs.get(cid, (err, block) => {
+          if (err) {
+            return cb(err)
+          }
+          const result = this.resolvers[cid.codec].resolver.resolve(block, path)
```

Comment:
there should be a check with a nice error message for `this.resolvers[cid.codec]` as this can easily be `undefined`